.travis.yml

```diff
@@ -38,7 +42,10 @@ env:
     - TASK=test:shared
 matrix:
   allow_failures:
+    - env: TASK=coverage:agent
```

Comment:
failed to manually produce coverage statistics for agent. Assumed it was allowed to fail for the same reason `coverage:finance` is in `allow_failures` (timeout) but observed this behaviour where the agent **unittests pass** without error but under instrumentation for **coverage** analysis four **tests fail** (therefore wrong coverage stats). 

Is this a known issue and the reason for `coverage:agent` being allowed to fail?

```  
 78 passing (53s)
   4 failing

   1) Contract: Agent app
        > Signing messages
          > Designated signer
            > Signature mode: ERC1271 -> EIP712
              isValidSignature returns true to a valid signature:
       Expected signature to be valid (returned 0x00000000)
       + expected - actual
       -0x00000000
       +0x20c13b0b
       
       at assertIsValidSignature (test/agent.js:268:14)
       at Context.it (test/agent.js:396:13)
       at process.internalTickCallback (internal/process/next_tick.js:77:7)
 
   2) Contract: Agent app
        > Signing messages
          > Designated signer
            > Signature mode: ERC1271 -> EIP712
              isValidSignature returns true to a valid signature with legacy version:
       Expected signature to be valid (returned 0x00000000)
       + expected - actual
       -0x00000000
       +0x20c13b0b
       
       at assertIsValidSignature (test/agent.js:268:14)
       at Context.it (test/agent.js:401:13)
       at process.internalTickCallback (internal/process/next_tick.js:77:7)
 
   3) Contract: Agent app
        > Signing messages
          > Designated signer
            > Signature mode: ERC1271 -> EthSign
              isValidSignature returns true to a valid signature:
       Expected signature to be valid (returned 0x00000000)
       + expected - actual
       -0x00000000
       +0x20c13b0b
       
       at assertIsValidSignature (test/agent.js:268:14)
       at Context.it (test/agent.js:396:13)
       at process.internalTickCallback (internal/process/next_tick.js:77:7)
 
   4) Contract: Agent app
        > Signing messages
          > Designated signer
            > Signature mode: ERC1271 -> EthSign
              isValidSignature returns true to a valid signature with legacy version:
       Expected signature to be valid (returned 0x00000000)
       + expected - actual
       -0x00000000
       +0x20c13b0b
       
       at assertIsValidSignature (test/agent.js:268:14)
       at Context.it (test/agent.js:401:13)
       at process.internalTickCallback (internal/process/next_tick.js:77:7)
 
 -------------------------|----------|----------|----------|----------|----------------|
 File                     |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
 -------------------------|----------|----------|----------|----------|----------------|
  contracts/              |    96.92 |       90 |    94.12 |    97.37 |                |
   Agent.sol              |    95.45 |    83.33 |       90 |    95.83 |            102 |
   SignatureValidator.sol |    97.67 |    91.67 |      100 |    98.08 |            126 |
  contracts/standards/    |       50 |      100 |       50 |       50 |                |
   ERC1271.sol            |       50 |      100 |       50 |       50 |             42 |
   IERC165.sol            |      100 |      100 |      100 |      100 |                |
 -------------------------|----------|----------|----------|----------|----------------|
 All files                |    95.52 |       90 |    89.47 |    96.15 |                |
 -------------------------|----------|----------|----------|----------|----------------|
```

Reply:
@tintinweb Yes, this is a known issue that we haven't been too concerned about yet (since the whole thing isn't _exactly_ ready and we're honestly a bit stumped at how this could be failing coverage). It does also timeout, because `solidity-coverage` goes crazy on its instrumentation.